### ISSUE TITLE
Support for Synology Photos

### ISSUE THREAD

**Issue by stepman0** [flingo64/PhotoStation-Upload-Lr-Plugin#48]
For testing purpose, I installed the current beta of Synology DSM 7 and the new Synolog Photos (in a Virtual Machine).

Unfortunately, the plugin does not work with the new version (as I expected ;-).

Are there any plans on supporting it?

**Comment by flingo64** [comment]
No plans yet, but definitely strongly interested!  I'm not registered for the DSM 7 Beta test program, so I cannot says how close the new Photos API is to the old Photo Station API. if someone could provide a Browser network capture (*.har file) of a typical Photos Browser session (login, browse a gallery, open a photo in detailed view) I could do a first check and compare of the API calls to estimate the upgrade efforts. 
Access to a Photos servers would be even better, but not neccessary for the first step.

**Comment by quizzical86** [comment]
It seems like there is a Photo's server that you can use: I just checked synology's own DSM7 demo platform and see that Synology Photo's is accessible there:
https://demo.synology.com/dsm

**Comment by flingo64** [comment]
meanwhile, stepman0 was so kind to support me with access to his DSM7/Photos Beta Preview.  Last week, I installed the DSM7/Photos as vDSM on my box, so testing is solved.

The plugin is almost ready for Photos, but there are still some real basic function in the Photos API that do not work reliable, so I can't get a stable result.

**Comment by quizzical86** [comment]
I just wanted to flag that DSM7 has been officially released on 29 June so it is now the official version :)
https://www.synology.com/en-uk/DSM70

**Comment by stepman0** [comment]
@flingo64 Have you already looked into the new and shiny DSM7. I updated without thinking of Photostation/Photos :-(

Any guess when your plugin will support Photos?

**Comment by flingo64** [comment]
I just installed the final DSM7 including Photos as VM to check if there are any changes to the Beta version.  My alpha-version plugin seems to work with all the restrictions encountered so far:
- overwriting photos is not part of the API, so when you re-publish a photo, the plugin will do a delete/upload, which doesn't work reliable: the old photo is sometimes not deleted but no longer visible in Photos, but the newly uploaded photo gets a filename suffix, which the plugin won't recognize and thus looses the correlation between the photo in Lr and Photos in the Published Collection. Although dissapearing photos is still an issue which can be solved vias re-indexing in Photos, the renaming issue seems to be fixed.
- the API does not allow to update metadata other than capture date or description/caption. Thus the MetadataUpload mode of the plugin is almost useless, because you will have to upload the photo whenever you just modified tags or GPS info
- for the same reason videos won't have the tags and GPS info assigned in Lr
- face regions defined in Lr are not supported by Photos
- location tags defined in Lr are not supported by Photos

Having said that I'm looking for two to three friendly users for the alpha version.

**Comment by stepman0** [comment]
I can test the alpha version.

**Comment by flingo64** [comment]
> 
> 
> I can test the alpha version.

Shipped to you via email!
:-)

**Comment by mathewsd001** [comment]
I'm also available for testing

**Comment by flingo64** [comment]
> 
> 
> I'm also available for testing

Shipped to you via email!
:-)

**Comment by Feilner** [comment]
@flingo64 May be you can create a branch on GitHub for the version to support  Synology Photos on DSM7. This would enable also others to test and also contribute.

**Comment by mathewsd001** [comment]
Hi Martin,
I am able to use Lightroom Classic with DSM 7 Photos by recreating my
structure in the Hard Drive Publish Services.
Regards,
dave

**Comment by flingo64** [comment]
Just installed Photos 1.1.0 0224 on my vDSM7 and the Photos app for Android. I'm really excited about the map view (at least in the Android app) - which was the killer feature of Photo Stations for me and now finally it's also available in Photos!
They also added the rating feature to Photos. So, they got me back on track! 
Stay tuned for a Xmas present ...

**Comment by quizzical86** [comment]
Without wanting to sound like a sad and lonely person (promise I'm not ;-P ) this might be the Christmas present I look forward to most! It's the last thing preventing me from migrating to DSM7.
I'm rather confused about how a migration from Photo Station to Photos would work for those using your great plugin; do you intend to provide some hints on that as well?

**Comment by fred-h20** [comment]
> Without wanting to sound like a sad and lonely person (promise I'm not ;-P ) this might be the Christmas present I look forward to most! It's the last thing preventing me from migrating to DSM7. I'm rather confused about how a migration from Photo Station to Photos would work for those using your great plugin; do you intend to provide some hints on that as well?

Same for me. I really want to update to DSM7, but I would miss this plugin really much. Thank you Martin for the great work within the last years and if we can support you with testing feel free to ask.

**Comment by flingo64** [comment]
>... I'm rather confused about how a migration from Photo Station to Photos would work for those using your great plugin; do you intend to provide some hints on that as well?

I'm currently concentrating on getting the plugin in a good shape, but I'm pretty sure there are guidelines from Synology how to  migrate the Photo Station to Photos. My expectation is that this migration process will migrate all photos/videos including thumbnails and add. videos to Photos and do a complete indexing thereafter. If that's done, you basically will have these changes in LR:
- in the Publish Service: 
    - change the Photo Server version from 'Photo Station 6.8' to 'Photos 1.1'
    - change the address to something like 'http://my-dsm7:5000' or 'https://my-dsm7:5001'
    - say 'yes' to let Lr move all photos to 'Modified to republish'
- for all Published Collection:
    - do a Publish with publsh mode 'CheckExisting' to make sure all photos were migrated successfully

If you want to build a copy of your gallery so you can use Photo Station and Photos in parallel (until you are sure Photos meets your expectations):
- create a duplicate of your Photo Station Publish Service with the exact same settings except the server version and address
- copy all Published collection and Publish Collection Sets from the old to the new Publish Service: use the export/import collection feature for that purpose
- **Very important: although exporting/importing will copy all collection settings, you still have to open them once and save them, otherwise the collection settings will not be effective**

**Comment by flingo64** [comment]
Photo StatLr 7.0.0 is online! It supports both Synology Photo Station and Synology Photos.

Please check the Release Notes or [Changelog](https://github.com/flingo64/PhotoStation-Upload-Lr-Plugin/blob/master/CHANGELOG.md) to get an understanding of how it works and which features are supported for Photos. There is also a short [FAQ on migrating from Photo Station to Photos](https://github.com/flingo64/PhotoStation-Upload-Lr-Plugin/blob/master/Documentation/40-Migrating-from-PhotoStation-to-Photos.md)

Thanks to stepman0 for supporting me in the first days of the Photos Beta and for alpha-testing and feedback!

This is my Christmas gift to the Photo StatLr community! More than 4.000 installations and still more than 200 active users (for a superb Photo Gallery that was discontinued by Synology!) were always a strong motivation to keep this thing going. Besides that, my own requirements and the fun working with the plugin helped a lot!

I wish you all a Merry Christmas and a lot of fun with Photo StatLr for Photos! Please, do not forget to pass on a Christmas gift to those who need our support, [donate to a good cause](https://messmer-online.de/index.php/software/donate-for-photo-statlr) and let other know you did it!

Martin

**Comment by quizzical86** [comment]
Upgrade going well so far, It's busy checking my 75000 photo's. Thanks a lot for the helpful instructions, wouldn't have been able to do it without them!

A tip for those who used the backup feature of DS Photo and want to use it in the new synology photo's app:
When selecting the backup destination, you will likely not be able to find the folder you used in DS Photo: you have to open Synology Photos using your computer and then go to settings -> shared space -> Set access permissions. There you need to enable mobile backup for the account you are using. Now you will be able to upload the photos you've taken with your mobile to the shared space folders (instead of only private folders which I don't use at all).

**Comment by quizzical86** [comment]
I've been struggling quite a bit with the check of the photo's, I ended up republishing a lot of them.
The issue I struggle most with is the error "17:48:31, ERROR: That does it, I'm leaving! Internal error: '[string "PSUploadTask.lua"]:123: attempt to index field 'exifTool' (a nil value)'"
Any clue what I could do to solve that?

**Comment by flingo64** [comment]
> I've been struggling quite a bit with the check of the photo's, I ended up republishing a lot of them.

7.1.0 fixes a lot of smaller issues

> The issue I struggle most with is the error "17:48:31, ERROR: That does it, I'm leaving! Internal error: '[string "PSUploadTask.lua"]:123: attempt to index field 'exifTool' (a nil value)'"

You found the very last bug!  ;-)  For the moment you can work around it by disabling the Upload options 'Color Label Tag' and 'Rating Tag' in the Published Collection settings. I already fixed it and will release it in 7.1.1. 
I opened an [new issue](#55) for it.
